msgraph-oauth-feature/src/com/microsoft/auth/OAuth2Feature.java

```diff
@@ -60,6 +60,8 @@ public boolean configure(FeatureContext context) {
       ctxt -> requestToken(ctxt, graphUri),
       graphUri
     );
+    oauth2.tokenScopeProperty(Property.SCOPE);
```

Comment:
it does not make sense to marriage these two APIs?

Reply:
I think the 'scopeProperty' is a little bit simpler to setup and will be used by many oauth2 features ... while the custom name is rarely used. So I thought it makes sense to make the most common case simple to setup.

Reply:
This could also be config property. A property that contains the name of a property that contains the scope. Is there a use case where this is different than PropertyScope?

Reply:
Today the Auth-Feautre developer has to use java code anyway, to glue his login-uris and auth-flow together. 
So I think having a factory to control the preferred setup isn't a bad thing. 
Writing rest-client or variables.yaml properties to control our implementations are IMHO harder to maintain and document. Like with the APIs, the user will not automatically know these properties. So we will have to build tooling to promote/configure them and add docs to it. With a java driven approach, the docs are easy to write and no extra tooling must be built. 

We can think of a good default setup here though, that would make this call obsolete, so just those that do not know our standard propty "AUTH.scope" will be forced to use this API.